cdci_data_analysis/flask_app/app.py

```diff
@@ -284,6 +284,8 @@ def run_analysis():
         logger.info("towards log_run_query_result")
         log_run_query_result(request_summary, r[0])
 
+        QueryOutJSON().load(r[0].json)
```

Comment:
Perhaps this is not complete? Since certain fields are not recognized?

Reply:
In principle, it does not need to specify all fields.
In fact, some fields can be optional.

Reply:
We should be aware that as this schema is derived from frontend, anything that does not fit it will not be shown correctly by frontend (and oda_api, which tries to follow it)

Reply:
But several tests are failing. Is there a way to "ignore" those unrecognized fields?

Reply:
At some point defaults for validation changed in marshmallow.
I made the necessary adaptations, all passes now.

But then it's even strange that all passes now. In fact, some fields are allowed by not used: e.g. error_message outside of exit status should be breaking validation.

We should eventually adapt schema to prevent these unused messages. I will make a simple attempt now.

Reply:
I made it so `error_message`, if present, must be empty. We should change the code where needed.

Reply:
suspiciously large number of them is failing. maybe it is expected. 
Could you please take over in verifying? 
If ok, we can close this one, and proceed with fixing in another one.

Reply:
yes thats ok